macos/Sources/Ghostty/SurfaceView_AppKit.swift

```diff
@@ -124,6 +124,9 @@ extension Ghostty {
         // A timer to fallback to ghost emoji if no title is set within the grace period
         private var titleFallbackTimer: Timer?
 
+        // The height of the titlebar in points, updated when window properties change
+        private var titlebarHeight: CGFloat = 0
```

Comment:
I don't think there's any reason to cache this. It seems fairly cheap to calculate on the fly, and that removes a whole lot of complexity around keeping it up to date during various events that I suspect we could get wrong at some point.

Reply:
I agree that caching the titlebar height isn't necessary. The calculation is very lightweight and removing the cache eliminates the complexity of keeping it in sync across various window events.

Regarding alternative approaches for detecting titlebar clicks, I've thought about it and here are some considerations:

   - Using hitTest might be less reliable since the titlebar area is part of our view when using hidden titlebar style
   
   - Adding a separate view for the titlebar would add complexity without clear benefits
   
   - Using window delegate methods would be too late as we need to intercept the click before window dragging

I'm open to exploring other approaches if you have specific ideas in mind!